internal/icinga2/api_responses.go

```diff
@@ -282,14 +282,22 @@ func UnmarshalEventStreamResponse(bytes []byte) (any, error) {
 	// decompositions in different variables will result in multiple manual fixes. Thus, a two-way deserialization
 	// was chosen which selects the target type based on the first parsed type field.
 
-	var responseType string
+	var (
+		responseType  string
+		responseError interface{}
```

Comment:
With `go1.22` we should prefer `any` over `interface{}`, I think :), but shouldn't the json key [`error`](https://github.com/Icinga/icinga2/blob/master/lib/remote/httputility.cpp#L67) always be of type `int`?

Reply:
I have not tested if it is the case here as well, but Icinga 2 doesn't send back json encoded error if the `accept` header is not set to `application/json`.

```bash
$ curl -kvsSiu root:icingas 'https://localhost:5665/v1/objects/hosts/docker-master?pretty=1' -H 'Accept: application/json'
---
* Closing connection
{"error":401,"status":"Unauthorized. Please check your user credentials."}%

$ curl -kvsSiu root:icingas 'https://localhost:5665/v1/objects/hosts/docker-master?pretty=1'
---
* Closing connection
<h1>Unauthorized. Please check your user credentials.</h1>%
```

Reply:
> With `go1.22` we should prefer `any` over `interface{}`, I think :)

:+1: 

> but shouldn't the json key [`error`](https://github.com/Icinga/icinga2/blob/master/lib/remote/httputility.cpp#L67) always be of type `int`?

I wasn't able to find this in the documentation, but this was also what I would expect. I will change it.

Reply:
Regarding the content not being JSON: in this case the HTTP status check should fire beforehand or, otherwise, the JSON unmarshalling should fail as HTML should not be able to be decoded as JSON.